torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -153,6 +153,28 @@ void initializeStreamsEvents(
   }
 }
 
+void initializeStreamsEvents(
+    std::vector<std::vector<at::Tensor>>& outputs,
+    std::vector<at::cuda::CUDAStream>& streams,
+    std::vector<at::cuda::CUDAEvent>& events) {
+  at::cuda::OptionalCUDAGuard guard;
+  streams.reserve(outputs.size());
+  events.resize(outputs.size());
+
+  for (size_t i = 0; i < outputs.size(); i++) {
```

Comment:
Can you add a check here for the assumption that all outputs in outputs[i] are on the same device?

Reply:
the new code does not have this assumption